M2/Macaulay2/m2/basictests/A06.m2

```diff
@@ -1,13 +1,6 @@
 -- test flatten
 assert := x -> if not x then error "assertion failed "
 
-
 assert({1,2,3,4,5} === flatten{1,{},{2},{3,4},5})
 f = { new MutableList from {0,1} }
 assert(f === flatten f)
-
--- test regexp
-assert( regex ( "^(.*[^ ])? *$", " abcdef " ) === {(0, 8), (0, 7)} )
-assert( regex ( "^ *(.*[^ ])? *$", " abcdef " ) === {(0, 8), (1, 6)} )
-assert( regex ( "^ *(.*)$", " abcdef " ) === {(0, 8), (1, 7)} )
-assert( regex ( ".?","" ) === {(0, 0)} )
```

Comment:
Two of these assertions still work, so why delete them.  The two failing ones point out a bug:
```m2
i34 : regex ( "^ *(.*)$", " abcdef " )

o34 = {(0, 8), (0, 8)}

o34 : List

i35 : regex ( "^( *)(.*)$", " abcdef " )

o35 = {(0, 8), (0, 1), (1, 7)}

o35 : List
```
The bug is that addition of parentheses changes the way things match -- in the first one, `.*` matches 8 characters, and in the second one, `.*` matches 7 characters.

It probably has to do with incorrect setting of flags, for the bug goes away if I specify none:
```m2
i36 : regex ( "^ *(.*)$", " abcdef " )

o36 = {(0, 8), (0, 8)}

o36 : List

i37 : regex ( "^ *(.*)$", " abcdef ", Flags => {} )

o37 = {(0, 8), (1, 7)}

o37 : List

i36 : regex ( "^ *(.*)$", " abcdef " )

o36 = {(0, 8), (0, 8)}

o36 : List

i37 : regex ( "^ *(.*)$", " abcdef ", Flags => {} )

o37 = {(0, 8), (1, 7)}

o37 : List
```

All 4 assertions should be restored.

Reply:
They're not deleted, simply moved to `tests/normal/regex.m2`.

Reply:
This is fixed in the latest commit. Lists are no longer accepted as input.
The assertions still exist and pass correctly, shall I mark this as resolved?

Reply:
I just tried it with you latest commit, and the bug is still present (that's what this conversation is about).  These answers should be the same:
```m2
i1 : (regex(".*(.*)","ab"))#1

o1 = (0, 2)

o1 : Sequence

i2 : (regex("(.*)(.*)","ab"))#2

o2 = (2, 0)

o2 : Sequence
```

Reply:
This is separate from your previous comment here, and we can keep talk about it in the regex PR thread. Shall I close this particular conversation? `Flags => {}` is no longer acceptable.